src/main/java/com/zavtech/morpheus/array/ArrayUtils.java

```diff
@@ -114,7 +114,8 @@
      * @return          the array of distinct values in the order they were observed
      */
     public static Array<Integer> distinct(IntStream values, int limit) {
-        final DistinctInts distinct = new DistinctInts(limit);
+        final int capacity = limit < it.unimi.dsi.fastutil.Arrays.MAX_ARRAY_SIZE ? limit : 1000;
```

Comment:
Should we throw a new IllegalArgumentException here if limit > MAX_ARRAY_SIZE? I'm wondering if it'll be confusing  users if we use a value that's  different than what they passed in